### PR TITLE
feat!: implement new activityV2 filter

### LINKED ISSUE #6044: Create activityV2 endpoints
Create new endpoints for filtering Tx data. The new endpoint should get data from the tables added in https://github.com/status-im/status-go/issues/6039

### PR DESCRIPTION
**Pull request by dlipicar**
Closes https://github.com/status-im/status-go/issues/6044

This PR creates v2 endpoints for the session-based activity API, which queries activity entries from the tables introduced in https://github.com/status-im/status-go/issues/6039 (only transactions sent from Status).

Noteworthy changes include:
- `StartActivityFilterSessionV2` API endpoint added, processing activity with the new logic.
- `transactions` field added to `Entry`, containing the list of transaction identities belonging to a multitransaction.
- `interactedContractAddress` field added to `Entry`, containing (when applicable, that is, for transactions other than a plain ETH transfer) the address of the smart contract the account interacted with in a given transaction.
- Transaction status updates are now sent over `EventActivityFilteringUpdate`
- The debounce mechanism when processing activity update events was improved. We now ensure the update is called 1 second after the initial trigger. Before, it could be delayed for very long if more events were received.
- We now properly finish the goroutine in charge of listening to activity update events. Before, goroutines were never properly killed.

### CODE REVIEW

File: services/wallet/activity/activity_v2.go [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+	"github.com/status-im/status-go/services/wallet/router/routes"
+	"github.com/status-im/status-go/services/wallet/token"
+	"github.com/status-im/status-go/services/wallet/transfer"
+	"github.com/status-im/status-go/sqlite"
+	"github.com/status-im/status-go/transactions"
+)
+
+// getActivityEntriesV2 queries the route_* and tracked_transactions based on filter parameters and arguments
+// it returns metadata for all entries ordered by timestamp column
+func getActivityEntriesV2(ctx context.Context, deps FilterDependencies, addresses []eth.Address, allAddresses bool, chainIDs []wCommon.ChainID, filter Filter, offset int, limit int) ([]Entry, error) {
```
Comment (saledjenic) [inline services/wallet/activity/activity_v2.go:30]: Do we need `ctx` param here?

File: services/wallet/activity/activity_v2.go [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+	if len(addresses) == 0 {
+		return nil, errors.New("no addresses provided")
```
Comment (saledjenic) [inline services/wallet/activity/activity_v2.go]: Does it make sense to have errors listed somewhere and just refer to appropriate?

File: services/wallet/activity/activity_v2.go [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+	return ret, nil
+}
+
+func getActivityTypeV2(processorName string, isApproval bool) Type {
+	if isApproval {
+		return ApproveAT
+	}
+
+	switch processorName {
+	case pathprocessor.ProcessorTransferName, pathprocessor.ProcessorERC721Name, pathprocessor.ProcessorERC1155Name:
```
Comment (saledjenic) [inline services/wallet/activity/activity_v2.go]: Do we need ens and stickers path processor names here?

File: services/wallet/activity/activity_v2.go [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+	}
+
+	return L2FinalizationDuration
+}
+
+func getTransferType(fromToken *token.Token, processorName string) *TransferType {
+	ret := new(TransferType)
+
+	switch processorName {
+	case pathprocessor.ProcessorTransferName:
```
Comment (saledjenic) [inline services/wallet/activity/activity_v2.go]: And here as well?

File: services/wallet/activity/session.go [hunk 5]
```diff
@@ -301,6 +332,7 @@ func (s *Service) GetMoreForFilterSession(id SessionID, pageCount int) error {
 	s.internalFilter(
```
Comment (saledjenic) [inline services/wallet/activity/session.go]: Adding a comment here, cause that is probably missed in some of the previous reviews, don't we need to add locking before the line `326`?

```go
    s.sessionsRWMutex.RLock()
    session, found := s.sessions[id]
    s.sessionsRWMutex.RUnlock()
```

**Review by saledjenic — APPROVED** [review]
I am not very well familiar with the code here but it looks ok to me, left a few minor comments.

**Review by smohamedjavid — APPROVED** [review]

File: services/wallet/activity/activity_v2.go (hunk 3 above)
Comment (dlipicar) [reply services/wallet/activity/activity_v2.go]: they're not handled in this initial iteration, there were no designs available until Ben added them a few days ago. Will implement support for those in a separate PR. They'll be shown as "Unidentified transaction" for now.

File: services/wallet/activity/activity_v2.go (hunk 4 above)
Comment (dlipicar) [reply services/wallet/activity/activity_v2.go]: same response

File: services/wallet/activity/activity_v2.go (hunk 2 above)
Comment (dlipicar) [reply services/wallet/activity/activity_v2.go]: will do!

File: services/wallet/activity/activity_v2.go (hunk 1 above)
Comment (dlipicar) [reply services/wallet/activity/activity_v2.go:30]: forgot to add it to the query, added it now

File: services/wallet/activity/session.go (hunk 5 above)
Comment (dlipicar) [reply services/wallet/activity/session.go]: ugh yes, re-checking all accesses to this map

**Pull request merged**